proposals/20250923-plugin-system-event-schema-versioning.md

```diff
@@ -0,0 +1,169 @@
+# Proposal for the introduction of driver event schema version check in the Plugin System
+
+## Summary
+
+This proposal introduces a comprehensive event schema version validation system for the Falco plugin framework. The system will provide runtime validation of the event schema version related to the events generated by the drivers, that can be consumed by plugins providing parsing and/or field extraction capabilities. This will ensure backward compatibility and clear error reporting for plugins that depend on specific schema versions.
```

Comment:
```suggestion
This proposal introduces a comprehensive event schema version validation system for the Falco plugin framework. The system will provide runtime validation of the event schema version related to events emitted by libscap, which can be consumed by plugins that provide parsing and/or field extraction capabilities. This will ensure backward compatibility and clear error reporting for plugins that depend on specific schema versions.
```